doc/developer-guides/trusty.rst

```diff
@@ -0,0 +1,186 @@
+#. Drivers for the Android kernel (Linux) to facilitate communication with applications
+   running under the Trusty OS
+#. A set of libraries for Android/Linux systems software to facilitate communication with
+   trusted applications executed within the Trusty OS using the kernel drivers
+
+LK (`Little Kernel`_) is a tiny operating system suited for small embedded devices, bootloaders,
+and other environments where OS primitives like threads, mutexes, and timers are needed, but
+there's a desire to keep things small and lightweight. LK has been chosen as the Trusty OS kernel.
+
+Trusty Architecture
+*******************
+
+Trusty Architectural diagram::
+
+                      +---------------------------+
+                      |VMn                        |
+                      |        ......             |
+    +------------+  +---------------------------+ |
+    |VM0         |  |VM1                        | |
+    |            |  | +--------+     +--------+ | |
+    |            |  | |        |     |        | | |
+    |    SOS     |  | | Normal |     | Secure | | |
+    |            |  | | World  |     | World  | |-+
+    |            |  | |        |     |        | |
+    |            |  | +--------+     +--------+ |
+    +------------+  +---------------------------+
+    +-------------------------------------------+
+    |             ACRN Hypervisor               |
+    +-------------------------------------------+
+    +-------------------------------------------+
+    |                   HW                      |
+    +-------------------------------------------+
+
+.. note::
+   Trusty OS is running in Secure World in the architecture above.
+
+Trusty specific Hypercalls
+**************************
+
+1. ``HC_INITIALIZE_TRUSTY``
+
+   * This Hypercall is used by the User OS bootloader (``UOS_Loader``) to request ACRN to initialize Trusty
+   * The Trusty memory region range, entry point must be specified
+   * Hypervisor needs to save current vCPU contexts (Normal World)
+#. ``HC_WORLD_SWITCH``
+
+   * Simulate ARM SMC (Secure Monitor Call, or SMC) instruction to do world switch
+   * Hypervisor needs to save current world vCPU contexts, and load next world vCPU contexts
+   * Update rdi, rsi, rdx, rbx to next world vCPU contexts
+
+APIs
+====
+
+.. code-block:: c
+
+   hcall_initialize_trusty(vm_t *vm)
+   hcall_world_switch(vm_t *vm)
+
+
+Trusty Boot flow
+****************
+
+By design, the User OS bootloader (``UOS_Loader``) will trigger the Trusty boot process. The complete bootflow is illustrated below.
+
+.. graphviz:: trusty/boot-flow.dot
+   :name: trusty-boot-flow
+   :align: center
+   :caption: Trusty Boot Flow
+
+Details
+
+1. UOS_Loader
+
+   a. Load and verify trusty image from virtual disk
+   #. Allocate runtime memory for trusty
+   #. Do ELF relocation of trusty image and get entry address
+   #. Call ``HC_INITIALIZE_TRUSTY`` with trusty memory base and entry address
```

Comment:
``HC_INITIALIZE_TRUSTY`` is defined as a number (in hypervisor/include/acrn_hv_defs.h) , so how do you call it?

Reply:
I suspect it should actually be ``hcall_initialize_trusty``, @YadongQi @dongyaozu @jren1 @anthonyzxu , can you confirm?